okio/src/main/java/okio/AbstractBufferedSource.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2017 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okio;
+
+import java.io.EOFException;
+import java.io.IOException;
+
+abstract class AbstractBufferedSource implements BufferedSource {
+
+  @Override public String readUtf8LineStrict() throws IOException {
+    return readUtf8LineStrict(Long.MAX_VALUE);
+  }
+
+  @Override public String readUtf8LineStrict(long limit) throws IOException {
+    if (limit < 0) throw new IllegalArgumentException("limit < 0: " + limit);
+    if (limit == 0) throw new EOFException("limit == 0");
+    long scanLength = limit == Long.MAX_VALUE ? Long.MAX_VALUE : limit + 1;
+    long newline = indexOf((byte) '\n', 0, scanLength);
+    Buffer buffer = buffer();
+    if (newline > -1) {
```

Comment:
nit: we typically write `!= -1` ’cause we’re comparing a true/false condition (was it found?) vs. a range question (was it at least 0?)